### PR TITLE
[2018.3.0rc1] fixes to x509 module and state for PY3

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
A couple changes to ensure that when we're using the contents of keys that we're trying them as strings and not bytes.

### What issues does this PR fix or reference?
#46716 

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cachedout** [comment]
@akissa Did you want to look at this? We usually have you take a peek at changes to this module. :]

**Comment by akissa** [comment]
I do not have py3 at the moment so i will not be able to test.

**Comment by rallytime** [comment]
Since this fixes a bug that @frogunder found, I have asked him to test this fix.

**Comment by frogunder** [comment]
@rallytime working fine for me now with this fix, both on py2 and py3.

File: salt/states/x509.py [hunk 1]
```diff
@@ -308,6 +308,9 @@ def private_key_managed(name,
         file_args['contents'] = __salt__['x509.create_private_key'](
             text=True, bits=bits, passphrase=passphrase, cipher=cipher, verbose=verbose)
 
+    # Ensure the key contents are a string before passing it along
+    file_args['contents'] = salt.utils.stringutils.to_str(file_args['contents'])
```
Comment (terminalmage) [inline salt/states/x509.py]: Is there a specific reason (like maybe x509 not working with unicode strings in PY2) that we're using `to_str()` here?

**Review by terminalmage — APPROVED** [review]
Just one question before merging, but otherwise this gets a :+1:

File: salt/states/x509.py (hunk 1 above)
Comment (garethgreenaway) [reply salt/states/x509.py]: Exactly that. It's expecting it to be a string and not bytes.

Comment (terminalmage) [reply salt/states/x509.py]: :+1:

**Pull request merged**